### ISSUE TITLE
Screen reader issues in the PSW

### ISSUE THREAD

**Issue by michaelasemple** [visitscotland/dot-com-brxm#64]
### Button labels are not clear when being read out by a screen reader:

When using the PSW to change the number of guests and units the buttons are announced as “hyphen” by voiceover on the iPad and “dash” by the JAWS screen reader on desktop depending on the level of punctuation the user has selected for their screen reader. The labels on buttons must be clear, such as “remove” or “add” so they tell the screen reader user what the function of the button does.

> When adding units, all buttons to add and remove adults, children, and infants are announced only as 'button',  these should be defined for each. (Separate note found in https://jira.visitscotland.com/browse/VSD-249)

**Given** a user is using a screen reader to navigate the PSW,
**When** they change the number of guests and units,
**Then** the labels on buttons must read as “remove” or “add”
 
Raised from RNIB A11y testing March 2023

### Some labels can be unclear in speech
The label for changing the units and guests in the PSW is “guests”. It is not clear in speech that this is the label for the element and in addition the number and the term it is associated with are not linked when interacting with the page using a screen-reader. For example, if you read the number 1 and then the word “unit” followed by the number 2 and then the word “guests”, it can be difficult in speech to know which number associates with which word. 
 
![guests label](https://github.com/visitscotland/dot-com-brxm/assets/143008024/f9f5fe23-750e-4477-8601-d81f4f002c64)

Another occasion where a screen reader user found the information confusing was the label above the calendar element. Because the task required to book accommodation for the 23rd May and the label said May 23 they initially thought this was the day not the year. Having the year as 2023 would have made this clearer.  

The “Add unit” text is a link and it can mislead the user into thinking that they are going to move to another page.

>Make sure labels and element names are clear and accurate. Numbers and words should be associated not only visually but also in speech.  

Raised from RNIB A11y testing March 2023
### No feedback for the type of element:

When the screen reader user was interacting with the guest element in the PSW, they did not receive any feedback whether this was an edit field or a pop up. The user thought they were not able to change the selection because of this lack of feedback for the element they had reached.  

- It must be clear if an element is interactive, then the type of element that it is must be announced by the screen reader for the user to know how to interact with it. 

Raised from RNIB A11y testing March 2023

### Screen reader announcement is confusing for users
Screen reader announcement for 'For' option is very clear by announcing 'For, combobox, accomodation, collapsed'. It follows the pattern:
Name- 'For'
Role -'Combobox'
Value -' Accommodation' + 'Collapsed'
For 'location' the screen reader announcement is:
"combobox collapsed location edit has autocomplete 'enter a location' blank
role value name"
Ideally the name, in this case "Location" would be announced first.
If the blank value is 'enter a location' then also announcing 'blank' could be confusing. This type of behaviour should be consistent across similar elements as it help screen reader users learn to navigate.

### Combo boxes
When accessing the combo boxes in the PSW - if the screen reader user tabs through them, at times, the associated label is not announced and the user must arrow through the form to know what they must insert.  
When tabbing through a form the visually associated label for the element must be announced by the screen reader. 
![image](https://github.com/visitscotland/dot-com-brxm/assets/143008024/b0669c68-eade-42a8-84fc-01edc7f382d1)
Raised from RNIB A11y testing March 2023

### Screen reader label needs defined for date pickers 'clear' link
Instead of saying 'clear link', 'clear dates selected' would be more informative because announcing this as a link rather than a button may be confusing.

**Given** a user has input dates within the accommodation OR events category in the PSW
**When** they choose to clear their selection 
**Then** the screen reader reads 'clear dates selected'

### Strange feedback for elements on mobile 

When interacting with the PSW, when there is a selection in the field, there is the option to clear it (unsure if this relates to the clear button or the x button to clear the field.) The feedback in speech for the clear button on the iPhone was “cross, times, diagram, gears”.  
The clear button must have a label that is clear and relevant to speech users.  

Raised from RNIB A11y testing March 2023
